商品一覧表示機能

shivonnu opened the pull request:
# WHAT
商品一覧表示機能を実装しました。
# WHY
最終課題フリマアプリの実装のため。

- sold out は購入機能実装後に実装します

- 以下Gyazo動画のリンクです
**修正した横並びになってる動画です
https://gyazo.com/537bcec19813d3d5ad81c7b8389c6ceb**

> こちらは修正前です
> https://gyazo.com/a0d1b912ccee075875ca1ac3b112a5dd

shivonnu committed:
fitst commit

新規投稿商品に商品出品ページリンクを作成、itemのコントローラーにインデックスメソッドを追加しました。

shivonnu committed:
画像、商品名、価格の表示

商品インスタンスから画像、商品名、価格の表示を展開しました。

shivonnu committed:
商品が無い場合のダミーを無い時だけの表示に変更,コメントアウト削除

出品した商品名がなければダミーが表示されないように変更、該当コメントアウトを削除しました。

shivonnu committed:
ログインしてない時にもインデックスページを見れるようにbefore_actionを移動

appricationコントローラーにあるとbefore_actionが全体に反映してしまうので、Userコントローラーに移動しました。move_to_indexは前工程に書いてましたが不要だったので削除しました。

Katsuno0814 commented on app/controllers/items_controller.rb:
```diff
@@ -1,7 +1,7 @@
-  #   @item = Item.all
-  # end
+  def index
+    @items = Item.all
```
こちら、商品を全て表示するだけでなく、orderメソッドを用いて、出品した順に並び替えましょう！

Katsuno0814 commented on app/views/items/index.html.erb:
```diff
@@ -112,18 +112,18 @@
-      <%# 商品のインスタンス変数になにか入っている場合、中身のすべてを展開できるようにしましょう %>
       <li class='list'>
-        <%= link_to "#" do %>
+        <% @items.each do |set| %>
```
こちらhttps://gyazo.com/a0d1b912ccee075875ca1ac3b112a5dd
見本アプリでは商品は横並びになっておりますので修正いたしましょう！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

shivonnu committed:
商品を出品した順に並び替える

orderメソッドを用いて商品を出品した順に並び替えました。

shivonnu committed:
タグの並び整備,each doの位置修正

横に並べられなすぎてタグを一通り整理して、下の方の</div>と</a>タグを不要なので削除しました。each do の位置をclass listより上にして横並びが反映されるように修正しました。

shivonnu commented:
コードレビューありがとうございます！
修正しましたので再度コードレビュー依頼させて頂きます！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！

shivonnu commented:
ありがとうございます！引き続き頑張ります！
